Benchmarks/Benchmarks/NIOCoreBenchmarks/NIOLoopBoundBox.swift

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOEmbedded
+import Benchmark
+
+func runNIOLoopBoundBoxInPlaceMutation(benchmark: Benchmark) {
+    let embeddedEventLoop = EmbeddedEventLoop()
+    let boundBox = NIOLoopBoundBox([Int](), eventLoop: embeddedEventLoop)
+    boundBox.value.reserveCapacity(1000)
+
+    benchmark.startMeasurement()
+
+    for _ in benchmark.scaledIterations {
+        boundBox.value.removeAll(keepingCapacity: true)
+        for i in 0..<1000 {
```

Comment:
We should only have the scaled iterations loop, not this inner one.

Reply:
How does that protect against runtime allocations?

Reply:
Also, this is now no longer testing what I wanted to test: the performance of repeated mutation of value. CoW types like Array generally can have one off allocations but that isn't the real problem. The problem is if it happens on every iteration in a loop.

Reply:
For the future, the reason looping `scaledIterations` times helps is that the malloc values output are divided by `scaledIterations`. This means any allocation that doesn't happen _every_ loop iteration is folded away, so spurious runtime allocations that may slightly differ between runs don't affect our thresholds.